packages/web3torrent/src/library/client-seeding.test.ts

```diff
@@ -74,38 +77,63 @@ describe('Seeding and Leeching', () => {
   it('should reach a ready-for-leeching, choked state', done => {
     seeder.seed(defaultFile as File, defaultSeedingOptions(), seededTorrent => {
       seeder.once(ClientEvents.PEER_STATUS_CHANGED, ({torrentPeers}) => {
-        expect(torrentPeers[`${leecher.pseAccount}`].allowed).toEqual(false);
+        expect(torrentPeers[`${leecherA.pseAccount}`].allowed).toEqual(false);
         done();
       });
-      leecher.add(seededTorrent.magnetURI, {store: MemoryChunkStore});
+      leecherA.add(seededTorrent.magnetURI, {store: MemoryChunkStore});
     });
   }, 10000);
 
-  it('should unchoke when seeder unblocks leecher', async done => {
-    await seeder.enable();
-    await leecher.enable();
-
+  it('should be able to unchoke and finish a download', async done => {
     seeder.seed(defaultFile as File, defaultSeedingOptions(), seededTorrent => {
       seeder.once(ClientEvents.PEER_STATUS_CHANGED, ({peerAccount}) => {
         seeder.once(ClientEvents.PEER_STATUS_CHANGED, ({torrentPeers}) => {
-          expect(torrentPeers[`${leecher.pseAccount}`].allowed).toEqual(true);
+          expect(torrentPeers[`${leecherA.pseAccount}`].allowed).toEqual(true);
         });
         seeder.togglePeer(seededTorrent.infoHash, peerAccount);
 
-        leecher.once(ClientEvents.TORRENT_DONE, ({torrent: leechedTorrent}) => {
+        leecherA.once(ClientEvents.TORRENT_DONE, ({torrent: leechedTorrent}) => {
           expect(seededTorrent.files[0].done).toEqual(leechedTorrent.files[0].done);
           expect(seededTorrent.files[0].length).toEqual(leechedTorrent.files[0].length);
           expect(seededTorrent.files[0].name).toEqual(leechedTorrent.files[0].name);
           done();
         });
       });
 
-      leecher.add(seededTorrent.magnetURI, {store: MemoryChunkStore});
+      leecherA.add(seededTorrent.magnetURI, {store: MemoryChunkStore});
     });
   }, 10000);
 
+  it('should support multiple leechers finishing their downloads', async done => {
+    const knownPeers = new Set();
+
+    seeder.seed(defaultFile as File, defaultSeedingOptions(), seededTorrent => {
+      seeder.on(ClientEvents.PEER_STATUS_CHANGED, ({peerAccount}) => {
+        // Make sure we toggle peers only twice
+        if (!knownPeers.has(peerAccount)) {
+          knownPeers.add(peerAccount);
+          seeder.togglePeer(seededTorrent.infoHash, peerAccount);
+        }
+      });
+      leecherA.add(seededTorrent.magnetURI, {store: MemoryChunkStore});
+      leecherB.add(seededTorrent.magnetURI, {store: MemoryChunkStore});
+    });
+
+    let finishCount = 0;
+    async function downloadFinished() {
+      finishCount += 1;
+      if (finishCount == 2) {
+        done();
+      }
+    }
+
+    leecherA.once(ClientEvents.TORRENT_DONE, downloadFinished);
+    leecherB.once(ClientEvents.TORRENT_DONE, downloadFinished);
```

Comment:
I don't really like putting more logic into the test cases. But typescript doesn't have event listener that can run n times only, e.g., `seeder.twice(EVENT, ...)`, and we decided to stick to the callback style tests, so 🤷‍♂ .

Reply:
Presumably you will eventually add an assertion on `finishCount`? Might it be nicer to keep a separate flag for each leecher? It might make debugging easier.

Reply:
Currently it calls `done()` when finishCount reaches 2, so it is equivalent to asserting on finishCount. 

Keeping a separate flag inside a leecher won't  get rid of this problem, because we still have to wait for both flag to turn true to finish the test.